src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -175,17 +185,28 @@ Alias Visitor::visitTypesafe(Parser::AliasWithoutBracketsContext* ctx) {
 }
 
 // ____________________________________________________________________________________
-Triples Visitor::visitTypesafe(Parser::ConstructQueryContext* ctx) {
+ParsedQuery Visitor::visitTypesafe(Parser::ConstructQueryContext* ctx) {
   if (!ctx->datasetClause().empty()) {
-    reportError(ctx, "Datasets are not supported");
+    reportError(ctx->datasetClause(0),
+                "QLever currently doesn't support FROM clauses");
   }
-  // TODO: once where clause is supported also process whereClause and
-  // solutionModifiers
+  ParsedQuery query;
+  // TODO: clean up
```

Comment:
Either clean it up, or specify what you need to clean up, e.g. "TODO<qup42> Set up the ParsedQuery at once using a constructor".